File: solutions/go/pangram/1/pangram.go
package pangram

import (
	"strings"
)

func IsPangram(input string) bool {
	charMap := map[rune]bool{}
	alph := "abcdefghijklmnopqrstuvwxyz"

	for _, char := range strings.ToLower(input) {
		if char >= 'a' && char <= 'z' {
			charMap[char] = true
		}
	}

	for _, char := range alph {
		if !charMap[char] {
			return false
		}
	}

	return true
}

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where
-- import Distribution.Simple.Utils (xargs)
import Data.Char (toLower)

isPangram :: String -> Bool
isPangram text
    | 26 == length [x | x <- ['a'..'z'], x `elem` lowertext] = True
    | otherwise = False
    where lowertext = map toLower text